cpp/src/arrow/python/helpers.cc

```diff
@@ -90,11 +90,11 @@ Status PythonDecimalToArrowDecimal(
   return FromString(c_string, arrow_decimal);
 }
 
-template Status ARROW_TEMPLATE_EXPORT PythonDecimalToArrowDecimal(
+template Status ARROW_EXPORT PythonDecimalToArrowDecimal(
     PyObject* python_decimal, decimal::Decimal32* arrow_decimal);
-template Status ARROW_TEMPLATE_EXPORT PythonDecimalToArrowDecimal(
+template Status ARROW_EXPORT PythonDecimalToArrowDecimal(
     PyObject* python_decimal, decimal::Decimal64* arrow_decimal);
-template Status ARROW_TEMPLATE_EXPORT PythonDecimalToArrowDecimal(
+template Status ARROW_EXPORT PythonDecimalToArrowDecimal(
     PyObject* python_decimal, decimal::Decimal128* arrow_decimal);
```

Comment:
Could this also be fixed by adding extern template declarations in helper.h, like we've done elsewhere?